### PR TITLE
calculated elements: enable within nested projections and subqueries

### PR DESCRIPTION
**Pull request by patricebender**
- **calculated elements in inline:** The base path,
i.e. the `column` with the `inline` property must be passed to the
`resolveCalculatedElement` function, as we need the `column.ref`
to be able to calculate whether the `inline` `refs` are join relevant
or not.
- **calculated elements in expand and subquery**: we must delay processing of subqueries to the very end of the column transformation. `cds.infer` attaches some properties, like the table alias of a join relevant path, to the refs in the calculated elements in the model. This is why we must process subqueries in the end, because otherwise, a subquery transformation might alter the helper properties in the model in a way, that they are not valid for the main query anymore.
- fix smart wildcard stuff

### CODE REVIEW

File: sqlite/test/deep/deepInputProcessing.test.js [hunk 1]
```diff
@@ -95,8 +95,11 @@ describe('UUID Generation', () => {
         })
     })
 
+    // in the query `select … { *, toOneChild { … } }` the expand actually replaces the
+    // the association `toOneChild` from the wildcard, hence `updated.toOneChild_ID === undefined`
+    // expect(updated.toOneChild.ID).toEqual(updated.toOneChild_ID)
```
Comment (patricebender) [inline sqlite/test/deep/deepInputProcessing.test.js:100]: e.g.:
```
entity MyEntity {
   key ID  : String;
       foo : Association to MyEntity;
       baz : String;
       bar : String;
}

entity foo as
   select from MyEntity {
      *,
      foo {bar} // replaces the assoc -> no foreign key
   };
```

File: db-service/test/cqn4sql/wildcards.test.js [hunk 2]
```diff
@@ -144,6 +210,60 @@ describe('wildcard expansion and exclude clause', () => {
         }
       `)
   })
+  it('xpr after wildcard replaces assoc from wildcard expansion', () => {
+    // "author.name as author" will replace the "author" association from Books -> no fk here
```
Comment (patricebender) [inline db-service/test/cqn4sql/wildcards.test.js:214]:
```suggestion
```

File: db-service/test/cqn4sql/wildcards.test.js [hunk 3]
```diff
@@ -144,6 +210,60 @@ describe('wildcard expansion and exclude clause', () => {
...
+          Books.dedication_addressee_ID,
+          Books.dedication_text,
+          Books.dedication_sub_foo,
+          Books.dedication_dedication,
+          Books.coAuthor_ID_unmanaged,
+        }
+      `)
+  })
+  it('val after wildcard replaces assoc from wildcard expansion', () => {
+    // "author.name as author" will replace the "author" association from Books -> no fk here
```
Comment (patricebender) [inline db-service/test/cqn4sql/wildcards.test.js:241]:
```suggestion
```

**Review by stewsk — APPROVED** [review]

**Comment by sjvans** [comment]
hi @patricebender @stewsk 

https://github.com/cap-js/cds-dbs/commit/0904b960eff44d07ae401ed9830ba7fedf551880#diff-3406bdae94230d4664b3a51d7efa5f3807f2a43095a9089ff018c5a5faf7ca3aR321 is breaking stuff. please check tests in dev setup.

thanks,
sebastian

**Comment by sjvans** [comment]
> hi @patricebender @stewsk
> 
> [0904b96#diff-3406bdae94230d4664b3a51d7efa5f3807f2a43095a9089ff018c5a5faf7ca3aR321](https://github.com/cap-js/cds-dbs/commit/0904b960eff44d07ae401ed9830ba7fedf551880#diff-3406bdae94230d4664b3a51d7efa5f3807f2a43095a9089ff018c5a5faf7ca3aR321) is breaking stuff. please check tests in dev setup.
> 
> thanks, sebastian

test to reproduce: https://github.com/cap-js/cds-dbs/pull/153

**Pull request merged**